spec/models/page_spec.rb

```diff
@@ -37,6 +37,12 @@
     expect(result.errors["position"]).to include("has already been taken")
   end
 
+  it "cannot duplicate Resource ID and Filename" do
+    result = described_class.create(filename: "04_ThirdPoint.xml", resource_id: 1, structure: structure, position: 1)
+
+    expect(result.errors["filename"]).to include("has already been taken")
```

Comment:
OK, but this also breaks the `position` validator, doesn't it?

Reply:
True, doesn't affect the test, but I fixed it anyhow to make sure we're targeting only filename